src/frontend/src/optimizer/logical_optimization.rs

```diff
@@ -254,6 +254,12 @@ lazy_static! {
         vec![AlwaysFalseFilterRule::create()],
         ApplyOrder::TopDown,
     );
+
+    static ref JOIN_PROJECT_TRANSPOSE: OptimizationStage = OptimizationStage::new(
+        "Join Project Transpose",
+        vec![JoinProjectTransposeRule::create()],
+        ApplyOrder::BottomUp,
+    );
```

Comment:
I think this optimization stage `JOIN_PROJECT_TRANSPOSE` can be applied together with `PROJECT_REMOVE`.